### DISCUSSION TITLE
showDisplay (Subtitles colum)

### DISCUSSION THREAD

**Discussion by xbette** [pymedusa/Medusa#10780] [Ideas & Feature requests]
it could be sweet if the "subtitles" colum shows the subtitles that is in the dir og a show. 

instead of what subtitles have been downloaded through medusa :) 

- so i mean, if i have downloaded subs from example subscene and put them into the show folder (9-1-1) then Medusa should see them and show what language is added :D

**Comment by p0psicles** [comment]
Can you provide an example of a subtitle name?
Just so I know the format.

**Reply by xbette** [reply]
My name template is this: 9-1-1.S02E01.Under.Pressure.1080p.WEB-DL.da.srt 

the (da) is for Danish :)

**Reply by p0psicles** [reply]
@xbette if you have subtitles enabled for that show, then medusa should already update the subtitle language for that episode.
You should see logs like these:
```
2022-06-26 10:39:36 DEBUG    SHOWQUEUE-REFRESH :: [ff56c8c] 1412: Subtitle changes detected for Arrow S08E10. Current subtitles: ['eng']
2022-06-26 10:39:40 DEBUG    SHOWQUEUE-REFRESH :: [ff56c8c] 1412: Saving subtitles changes to database
2022-06-26 10:39:53 DEBUG    SHOWQUEUE-REFRESH :: [ff56c8c] 1412: Saving episode to database: Arrow S08E10
2022-06-26 10:39:53 DEBUG    SHOWQUEUE-REFRESH :: [ff56c8c] 1412: Not creating SQL query - record is not dirty
```

**Reply by p0psicles** [reply]
was able to scan this from the filename you provided: 
![image](https://user-images.githubusercontent.com/1331394/175806697-a45f0202-0a2b-43cf-8145-e6426d52c41a.png)

**Reply by xbette** [reply]
oooh, it makes sense now for me :D 

i thought that when subs is enabled it just searched for subs on the internet and downloaded them. but it actually sees the subs that is merged into the file. AND the seperated subs (.srt files) So everything is how i wanted it. Thanks :)

**Comment by xbette** [comment]
Is it possible that you can do so it also regonize the language if the language part is "danish" or "english" if you understand? 

![26-06-2022 16-05-04](https://user-images.githubusercontent.com/3235438/175818069-0bae5732-e91a-4563-910c-6d9045506931.png)


![26-06-2022 16-08-02](https://user-images.githubusercontent.com/3235438/175818154-53655bc5-d702-4986-b6c8-647e4b4e751a.png)

**Comment by p0psicles** [comment]
@xbette i've looked into the possibility of using Guessit to parse the subtitle. But guessit doesn't recognize `Danish` as a valid subtitle.
```
For: Arrow - S08E10 - Fadeout - 1080p HDTV.Danish.srt
GuessIt found: {
    "title": "Arrow",
    "season": 8,
    "episode": 10,
    "episode_title": "Fadeout",
    "screen_size": "1080p",
    "source": "HDTV",
    "release_group": "Danish",
    "container": "srt",
    "mimetype": "text/plain",
    "type": "episode"
}
```
But it sees it as a release group.
guessit would work for other `valid` subtitle codes. But so would the current solution. So I don't see much added value there.

**Reply by p0psicles** [reply]
I think i found another workaround. Put it in a PR>
https://github.com/pymedusa/Medusa/pull/10782

**Reply by xbette** [reply]
Alright

**Reply by p0psicles** [reply]
hmm thinking about it. i'm now manually converting Danish -> dan. but you actually would want to have a more generic solution.
So then you'd need to add a convertion for every language possible.

**Reply by p0psicles** [reply]
Okay, found A more universal solution.
Now it should also accept all language names. Like English, German, Dutch, etc.

**Reply by xbette** [reply]
:D that sounds good!

**Reply by p0psicles** [reply]
okay, merged to develop

**Reply by xbette** [reply]
i see.. Thanks :D
